src/mem/ruby/common/DataBlock.hh

```diff
@@ -94,6 +106,9 @@ class DataBlock
     void alloc();
     uint8_t *m_data;
     bool m_alloc;
+
+    // Tracks block changes when atomic ops are applied
+    std::deque<uint8_t*> m_atomicLog;
```

Comment:
Wouldn't it make sense to use a unique_ptr instead?

Reply:
The DataBlock can be copied so doesn't this just make things more complex?